src/Integrations/ActiveCampaign/ActiveCampaign.php

```diff
@@ -0,0 +1,394 @@
+		$ssr = (bool) ($formAdditionalProps['ssr'] ?? false);
+
+		return $this->buildForm(
+			$this->getFormFields($formId, $ssr),
+			\array_merge($formAdditionalProps, $this->getFormAdditionalProps($formId, $type))
+		);
+	}
+
+	/**
+	 * Get mapped form fields.
+	 *
+	 * @param string $formId Form Id.
+	 * @param bool $ssr Does form load using ssr.
+	 *
+	 * @return array<int, array<string, mixed>>
+	 */
+	public function getFormFields(string $formId, bool $ssr = false): array
+	{
+		// Get item Id.
+		$itemId = $this->getSettingsValue(SettingsActiveCampaign::SETTINGS_ACTIVE_CAMPAIGN_LIST_KEY, (string) $formId);
+		if (empty($itemId)) {
+			return [];
+		}
+
+		// Get fields.
+		$fields = $this->activeCampaignClient->getItem($itemId);
+		if (empty($fields)) {
+			return [];
+		}
+
+		return $this->getFields($fields, $formId, $ssr);
+	}
+
+	/**
+	 * Map ActiveCampaign fields to our components.
+	 *
+	 * @param array<string, mixed> $data Fields.
+	 * @param string $formId Form ID.
+	 * @param bool $ssr Does form load using SSR.
+	 *
+	 * @return array<int, array<string, mixed>>
+	 */
+	private function getFields(array $data, string $formId, bool $ssr): array
+	{
+		$output = [];
+
+		// Bailout if data is empty.
+		if (!$data) {
+			return $output;
+		}
+
+		// Find fields.
+		$fields = $data['fields'] ?? [];
+
+		// Bailout if fields are empty.
+		if (!$fields) {
+			return $output;
+		}
+
+		// Loop fields and map.
+		foreach ($fields as $field) {
+			if (empty($field)) {
+				continue;
+			}
+
+			$type = $field['type'] ?? '';
+			$name = $field['name'] ?? '';
+			$label = $field['label'] ?? '';
+			$header = $field['header'] ?? '';
+			$required = $field['isRequired'] ?? false;
+			$options = $field['options'] ?? [];
+			$id = $field['id'] ?? '';
+
+			if (!$name) {
+				$name = $id;
+			}
+
+			// Some fields will not have label so use header.
+			if (!$label) {
+				$label = $header;
+			}
+
+			switch ($type) {
+				case 'firstname':
+					$output[] = [
+						'component' => 'input',
+						'inputName' => 'firstName',
+						'inputTracking' => 'firstName',
+						'inputFieldLabel' => $label,
+						'inputId' => 'firstName',
+						'inputType' => 'text',
+						'inputIsRequired' => $required,
+						'blockSsr' => $ssr,
+					];
+					break;
+				case 'lastname':
+					$output[] = [
+						'component' => 'input',
+						'inputName' => 'lastName',
+						'inputTracking' => 'lastName',
+						'inputFieldLabel' => $label,
+						'inputId' => 'lastName',
+						'inputType' => 'text',
+						'inputIsRequired' => $required,
+						'blockSsr' => $ssr,
+					];
+					break;
+				case 'fullname':
+					$output[] = [
+						'component' => 'input',
+						'inputName' => 'fullName',
+						'inputTracking' => 'fullName',
+						'inputFieldLabel' => $label,
+						'inputId' => 'fullName',
+						'inputType' => 'text',
+						'inputIsRequired' => $required,
+						'blockSsr' => $ssr,
+					];
+					break;
```

Comment:
If the capitalization of `inputName` is not relevant, I think it would be a good idea to use a single case for such similar output readability-wise.

Reply:
you are referring to this `fullName`?

Reply:
yes, `fullName` and `firstName` etc. if you can use `$type` as `inputName` it makes sense for this to be a single case.

Reply:
API defines that it needs to be provided like this